test/e2e/upgrade_management_cluster_test.go

```diff
@@ -0,0 +1,352 @@
+
+	By("Install Ironic in the target cluster")
+	installIronicBMO(upgradeClusterProxy, "true", "false")
+
+	By("Restore original BMO configmap")
+	restoreBMOConfigmap()
+	By("Reinstate Ironic Configmap")
+	configureIronicConfigmap(false)
+
+	By("Add labels to BMO CRDs in the target cluster")
+	labelBMOCRDs(upgradeClusterProxy)
+
+	By("Ensure API servers are stable before doing move")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return bootstrapClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert bootstrap API server stability")
+	Consistently(func() error {
+		kubeSystem := &corev1.Namespace{}
+		return upgradeClusterClient.Get(ctx, client.ObjectKey{Name: "kube-system"}, kubeSystem)
+	}, "5s", "100ms").Should(BeNil(), "Failed to assert target API server stability")
+	By("Moving the cluster to self hosted")
+
+	cmd := exec.Command(clusterctlBinaryPath, "move", "--to-kubeconfig", upgradeClusterProxy.GetKubeconfigPath(), "-n", namespace, "-v", "10")
+	output, err := cmd.CombinedOutput()
+	Logf("move: %v", string(output))
+	Expect(err).To(BeNil())
+
+	By("Check if BMH is in provisioned state")
+	Eventually(func() error {
+		bmhList := &bmo.BareMetalHostList{}
+		if err := upgradeClusterClient.List(ctx, bmhList, client.InNamespace(namespace)); err != nil {
+			return err
+		}
+		for _, bmh := range bmhList.Items {
+			if !bmh.WasProvisioned() {
+				return errors.New("BMHs cannot be provisioned")
+			}
+		}
+		return nil
+	}, e2eConfig.GetIntervals(specName, "wait-object-provisioned")...).Should(BeNil())
+	// TODO (Mohammed) after merging template that can be parameterized from the e2e the cluster should boot with 1 kcp and remove this scale in part
+	Logf("Scale down KCP to 1 in order to 2 VMs for the workload cluster")
+	cmd = exec.Command("kubectl", "scale", "kcp", clusterName, "-n", namespace, "--replicas", "1", "--kubeconfig", upgradeClusterProxy.GetKubeconfigPath())
+	output, err = cmd.CombinedOutput()
+	Logf("scale: %v", string(output))
+	Expect(err).To(BeNil())
+
+	Logf("Waiting for 2 BMHs to be in Available state")
+	Eventually(func(g Gomega) {
+		bmhs, err := getAllBmhs(ctx, upgradeClusterClient, namespace, specName)
+		g.Expect(err).To(BeNil())
+		g.Expect(filterBmhsByProvisioningState(bmhs, bmo.StateAvailable)).To(HaveLen(2))
+	}, e2eConfig.GetIntervals(specName, "wait-machine-remediation")...).Should(Succeed())
+
+	/*-------------------------------*
+	| Create a test workload cluster |
+	*--------------------------------*/
+	var (
+		controlPlaneMachineCount = pointer.Int64Ptr(1)
+		workerMachineCount       = pointer.Int64Ptr(1)
+	)
+	By("Creating a test workload cluster")
+
+	Logf("Creating the workload cluster with name %q using the %q template (Kubernetes %s, %d control-plane machines, %d worker machines)",
+		workloadClusterName, "(default)", kubernetesVersion, *controlPlaneMachineCount, *workerMachineCount)
```

Comment:
Why not to directly use the 'default' in the log?
```suggestion
	Logf("Creating the workload cluster with name %q using the default template (Kubernetes %s, %d control-plane machines, %d worker machines)",
		workloadClusterName, kubernetesVersion, *controlPlaneMachineCount, *workerMachineCount)
```